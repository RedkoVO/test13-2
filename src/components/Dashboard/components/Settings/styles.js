export default () => ({
  root: {
    paddingTop: '35px',
    paddingBottom: '250px'
  },
  title: {
    fontSize: '37px',
    fontWeight: 700,
    color: '#5e7cff',
    textAlign: 'center',
    marginBottom: '0px',
    textDecoration: 'none',
    margin: '0 auto',
    paddingBottom: '35px'
  },
  themeTitle: {
    textAlign: 'center'
  },
  wrBtns: {
    textAlign: 'center'
  },
  btn: {
    color: '#f0f0f0',
    padding: '5px 20px',
    fontSize: '12px',
    marginTop: '10px',
    fontWeight: 700,
    borderRadius: '15px',
    textDecoration: 'none',
    backgroundColor: '#5e7cff',
    cursor: 'pointer',

    '&:focus': {
      outline: 'none'
    }
  },
  disableBtn: {
    marginLeft: '14px',
    backgroundColor: '#a9a9a9'
  }
})
