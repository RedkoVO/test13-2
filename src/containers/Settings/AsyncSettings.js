import Async from '../../components/App/components/Async'

export const AsyncSettings = Async({
  loader: () => import(/* webpackChunkName: "settings_page" */ './index')
})

export default AsyncSettings
