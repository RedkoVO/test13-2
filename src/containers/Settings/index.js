import compose from 'recompose/compose'

import AsyncSettingsDesktop from '../../components/Settings/Desktop/AsyncSettingsDesktop'

export default compose()(AsyncSettingsDesktop)