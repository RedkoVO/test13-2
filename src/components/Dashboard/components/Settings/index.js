import React from 'react'
import PropTypes from 'prop-types'
import Checkbox from '@material-ui/core/Checkbox'
import withStyles from '@material-ui/core/styles/withStyles'
import cn from 'classnames'

import styles from './styles'

const Settings = ({
  classes,
  handleSubscribeCheckbox,
  isSubscribeCheckbox
}) => {
  return (
    <section id="settings">
      <div className={classes.root}>
        <div className={classes.title}>Settings</div>
        <div className={classes.themeTitle}>Theme</div>
        <div className={classes.wrBtns}>
          <button className={classes.btn}>Light</button>
          <button className={cn(classes.btn, classes.disableBtn)}>Dark</button>
        </div>

        <div className={classes.wrCheckbox}>
          <Checkbox
            checked={isSubscribeCheckbox}
            onChange={() => handleSubscribeCheckbox()}
            value="checkedB"
            color="primary"
          />
          Subscribe for funds receiving
        </div>
      </div>
    </section>
  )
}

Settings.propTypes = {
  handleSubscribeCheckbox: PropTypes.func,
  isSubscribeCheckbox: PropTypes.bool
}

export default withStyles(styles)(Settings)
