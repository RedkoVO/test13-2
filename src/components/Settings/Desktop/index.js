import React from 'react'
import withStyles from '@material-ui/core/styles/withStyles'

import styles from './styles'

const Settings = ({ classes }) => (
  <div className={classes.root}>
    Settings
  </div>
)

export default withStyles(styles)(Settings)