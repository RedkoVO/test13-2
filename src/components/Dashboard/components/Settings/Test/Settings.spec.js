import * as React from 'react'
import renderer from 'react-test-renderer'

import Settings from '../'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Settings handleSubscribeCheckbox={() => {}} isSubscribeCheckbox={true} />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
