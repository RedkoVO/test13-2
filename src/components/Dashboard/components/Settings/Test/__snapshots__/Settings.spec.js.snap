// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  id="settings"
>
  <div
    className="Settings-root-1"
  >
    <div
      className="Settings-title-2"
    >
      Settings
    </div>
    <div
      className="Settings-themeTitle-3"
    >
      Theme
    </div>
    <div
      className="Settings-wrBtns-4"
    >
      <button
        className="Settings-btn-5"
      >
        Light
      </button>
      <button
        className="Settings-btn-5 Settings-disableBtn-6"
      >
        Dark
      </button>
    </div>
    <div>
      <span
        className="MuiButtonBase-root-23 MuiIconButton-root-17 MuiPrivateSwitchBase-root-13 MuiCheckbox-root-7 MuiCheckbox-colorPrimary-11 MuiPrivateSwitchBase-checked-14 MuiCheckbox-checked-8"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label-22"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-26"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
            />
          </svg>
          <input
            checked={true}
            className="MuiPrivateSwitchBase-input-16"
            data-indeterminate={false}
            onChange={[Function]}
            type="checkbox"
            value="checkedB"
          />
        </span>
        <span
          className="MuiTouchRipple-root-35"
        />
      </span>
      Subscribe for funds receiving
    </div>
  </div>
</section>
`;
